web/src/app/modules/notification/modules/modals/components/typed-modal-content.component.ts

```diff
@@ -27,15 +30,19 @@ export class TypedModalContent extends ModalBase {
         private dataService: SpecmateDataService,
         public navigator: NavigatorService,
         private translate: TranslateService) {
-            super(activeModal);
-            this._options = Dialogtype.DEFAULT;
-        }
+        super(activeModal);
+        this._options = Dialogtype.DEFAULT;
+    }
 
 
-    public save(): void {
+    public async save() {
         if (this.isSaveShown) {
+            let element = this.navigator.currentElement;
+            if (Type.is(element, CEGModel) || Type.is(element, Process)) {
+                await this.dataService.saveModelImage(element);

```

Comment:
I find it strange that the code to generate the model image is in a modal diaglogue. In my opinion a modal dialoge should not contain such a logic